src/main/kotlin/pt/up/fe/ni/website/backend/service/PostService.kt

```diff
@@ -0,0 +1,36 @@
+package pt.up.fe.ni.website.backend.service
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import pt.up.fe.ni.website.backend.model.Post
+import pt.up.fe.ni.website.backend.repository.PostRepository
+
+class PostDto {
+    var title: String? = null
+    var body: String? = null
+    var thumbnailPath: String? = null
+}
+
+@Service
+class PostService(private val repository: PostRepository) {
+    fun getAllPosts(): List<Post> = repository.findAll().toList()
+
+    fun getPost(PostId: Long): Post =
+        repository.findByIdOrNull(PostId.toLong()) ?: throw NoSuchElementException("Post Not Found")
```

Comment:
```suggestion
        repository.findByIdOrNull(PostId) ?: throw NoSuchElementException("Post Not Found")
```

Reply:
My bad, I thought I had deleted that already.